internal/clients/auth.go

```diff
@@ -22,10 +23,13 @@ func NewResourceManagerAccount(ctx context.Context, config authentication.Config
 	objectId := ""
 
 	// TODO remove this when we confirm that MSI no longer returns nil with getAuthenticatedObjectID
+	// todo comment out for now as it is not stack env aware, add in a env param for it to use so it doens't look it up?
 	if getAuthenticatedObjectID := config.GetAuthenticatedObjectID; getAuthenticatedObjectID != nil {
```

Comment:
Are you suggesting that I should add a property(envvar) for configuration that checks if you are using ADFS environment or not? Otherwise, how can you know if you are using ADFS enviroment?

Reply:
@tombuildsstuff  cc @thetonymaster

Reply:
yes to fix ADFS you are going to need to be able to determine when your re in an ADFS env and then correctly handle this condition

Reply:
Thanks  @katbyte and @tombuildsstuff -- we haven't been able to find a reliable way to determine whether we're on ADFS or not. A suggestion by @coderGo93 is to set an envvar but that sounds not ideal. Any suggestions based on your experience?